Description: authPrompt: set value of beginRequestType to 'DONT_PROVIDE_USERNAME'
Author: yun341 <5933468@qq.com>
From: Matthew Ruffell <matthew.ruffell@canonical.com>
Origin: https://gitlab.gnome.org/GNOME/gnome-shell/-/commit/3fb321fd214469130539e1e31461257a749bff21
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1886592
Bug-GNOME-Merge: https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1443
---
 js/gdm/authPrompt.js | 2 ++
 1 file changed, 2 insertions(+)

Index: gnome-shell-3.36.4/js/gdm/authPrompt.js
===================================================================
--- gnome-shell-3.36.4.orig/js/gdm/authPrompt.js
+++ gnome-shell-3.36.4/js/gdm/authPrompt.js
@@ -7,6 +7,7 @@ const Animation = imports.ui.animation;
 const Batch = imports.gdm.batch;
 const GdmUtil = imports.gdm.util;
 const OVirt = imports.gdm.oVirt;
+const Vmware = imports.gdm.vmware;
 const Params = imports.misc.params;
 const ShellEntry = imports.ui.shellEntry;
 const UserWidget = imports.ui.userWidget;
@@ -484,6 +485,7 @@ var AuthPrompt = GObject.registerClass({
             // respond to the request with the username
             beginRequestType = BeginRequestType.PROVIDE_USERNAME;
         } else if (this._userVerifier.serviceIsForeground(OVirt.SERVICE_NAME) ||
+                   this._userVerifier.serviceIsForeground(Vmware.SERVICE_NAME) ||
                    this._userVerifier.serviceIsForeground(GdmUtil.SMARTCARD_SERVICE_NAME)) {
             // We don't need to know the username if the user preempted the login screen
             // with a smartcard or with preauthenticated oVirt credentials
